process/block/baseProcess.go

```diff
@@ -1294,19 +1294,55 @@ func (bp *baseProcessor) commitTrieEpochRootHashIfNeeded(metaBlock *block.MetaBl
 	}
 
 	balanceSum := big.NewInt(0)
+	nrAccounts := 0
+	nrAccountsWithDataTrie := 0
+	nrNotAccounts := 0
+	totalSizeAccounts := 0
+	totalSizeAccountsDataTries := 0
+	totalSizeNonAccounts := 0
 	for leaf := range allLeavesChan {
 		userAccount, errUnmarshal := unmarshalUserAccount(leaf.Key(), leaf.Value(), bp.marshalizer)
 		if errUnmarshal != nil {
+			nrNotAccounts++
+			totalSizeNonAccounts += len(leaf.Value())
 			log.Trace("cannot unmarshal user account. it may be a code leaf", "error", errUnmarshal)
 			continue
 		}
+
+		rh := userAccount.GetRootHash()
+		if len(rh) != 0 {
+			dataTrie, err2 := userAccountsDb.GetAllLeaves(rh)
```

Comment:
This will add a very large overhead. Consider having another debug flag for this.
Also, might rename `err2` to `errDataTrieGet`

Reply:
fixed